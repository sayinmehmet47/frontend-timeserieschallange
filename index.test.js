const { render } = require('@testing-library/react');
import { screen, configure, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const { default: Home } = require('./pages');
const { default: TimeChart } = require('./pages/components/TimeChart');
const router = require('next/router');

describe('Assets', function () {
  it('renders without crashing', () => {
    render(<Home />);
    expect(screen.getByText('ASSETS')).toBeInTheDocument();
  });
  it('should have cards with selection asset', () => {
    const { container } = render(<Home />);
    expect(container.querySelectorAll('.card-group')).toHaveLength(1);
    expect(container.querySelectorAll('.card-group .card')).toHaveLength(3);
  });

  it('first card should have two buttons', () => {
    const { container } = render(<Home />);
    expect(
      container.querySelector('.card ').querySelectorAll('button')
    ).toHaveLength(2);
  });
  it('view on map button should take me to google map page', () => {
    render(<Home />);

    //find all buttons with text "View on Map"
    const buttons = screen.getAllByText('View on Map ğŸ—º');
    buttons.forEach((button) => {
      expect(button).toBeInTheDocument();
    });
    //click on button calls router.push
    buttons.forEach((button) => {
      router.push = jest.fn();
      fireEvent.click(button);
      expect(router.push).toHaveBeenCalled();
    });
  });
  it('card should have button with text View Signals', () => {
    render(<Home />);
    const buttons = screen.getAllByText('View Signals ğŸ“¡');
    expect(buttons).toHaveLength(3);
    buttons.forEach((button) => {
      expect(button).toBeInTheDocument();
      router.push = jest.fn();
      fireEvent.click(button);
      expect(router.push).toHaveBeenCalledTimes(1);
    });
  });
});

describe('Signals', function () {
  const signals = [
    {
      SignalName: 'Signal 1',
      SignalMeasurement: [
        [[1636299582230, 114.1636]],
        [1636299690273, 114.3345],
        [1636299763970, 114.1636],
      ],
    },
  ];

  it('should graph render without crash', () => {
    render(<TimeChart signals={signals} />);
    expect(screen.getByText(/Chart with 3 data points./i));
  });
});
