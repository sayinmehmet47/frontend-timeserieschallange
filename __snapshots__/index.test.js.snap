// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets should have buttons with view on map and view signals 1`] = `
<div>
  <h1
    class="text-center mt-3"
  >
    ASSETS
  </h1>
  <div
    class="row card-group"
  >
    <div
      class="col-sm-4 g-5"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="text-success card-title"
          >
            <h4>
              UW Beznau
            </h4>
          </div>
          <p
            class="card-text"
          >
            Longitude:
            8.2338560914
          </p>
          <p
            class="card-text"
          >
            Latitude:
            47.5568277613
          </p>
          <div
            class="d-flex justify-content-between"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              View on Map ðŸ—º
            </button>
            <button
              class="btn-danger btn btn-secondary"
              type="button"
            >
              View Signals ðŸ“¡
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-sm-4 g-5"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="text-success card-title"
          >
            <h4>
              UW Grynau
            </h4>
          </div>
          <p
            class="card-text"
          >
            Longitude:
            8.9728980141
          </p>
          <p
            class="card-text"
          >
            Latitude:
            47.219215463
          </p>
          <div
            class="d-flex justify-content-between"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              View on Map ðŸ—º
            </button>
            <button
              class="btn-danger btn btn-secondary"
              type="button"
            >
              View Signals ðŸ“¡
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-sm-4 g-5"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="text-success card-title"
          >
            <h4>
              UW Winkeln
            </h4>
          </div>
          <p
            class="card-text"
          >
            Longitude:
            9.3026581653
          </p>
          <p
            class="card-text"
          >
            Latitude:
            47.397979458
          </p>
          <div
            class="d-flex justify-content-between"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              View on Map ðŸ—º
            </button>
            <button
              class="btn-danger btn btn-secondary"
              type="button"
            >
              View Signals ðŸ“¡
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
